ydb/core/tx/datashard/prefix_kmeans.cpp

```diff
@@ -494,28 +470,100 @@ class TPrefixKMeansScan final: public TPrefixKMeansScanBase, private TCalculatio
             cells[0] = TCell::Make(Parent);
             TSerializedCellVec::UnsafeAppendCells(cells, pk);
             PrefixBuf.AddRow(TSerializedCellVec{std::move(pk)}, TSerializedCellVec::Serialize({}));
-        } else if (!TCellVectorsEquals{}(Prefix.GetCells(), key.subspan(0, PrefixColumns))) {
-            HasNextPrefix = true;
-            return EScan::Reset;
         }
-        ++ReadRows;
-        ReadBytes += CountBytes(key, row);
-        switch (State) {
-            case EState::SAMPLE:
-                return FeedSample(row);
-            case EState::KMEANS:
-                return FeedKMeans(row);
-            case EState::UPLOAD_BUILD_TO_BUILD:
-                return FeedUploadBuild2Build(key, row);
-            case EState::UPLOAD_BUILD_TO_POSTING:
-                return FeedUploadBuild2Posting(key, row);
-            default:
-                Y_ASSERT(false);
-                return EScan::Final;
+
+        if (IsFirstPrefixFeed && IsPrefixRowsValid) {
```

Comment:
We don't need `IsFirstPrefixFeed`, we have `Sample` state, you can just put this code inside FeedSample

Reply:
https://github.com/ydb-platform/ydb/pull/16282#discussion_r2016877208